test/with_dummyserver/test_https.py

```diff
@@ -1035,7 +1036,53 @@ def test_http2_probe_no_result_in_ssl_error(self) -> None:
                     pool.request("GET", "/", retries=False)
 
             # The probe was inconclusive since an error occurred during connection.
-            assert http2_probe.values() == {(self.host, self.port): None}
+            assert http2_probe._values() == {(self.host, self.port): None}
+        finally:
+            urllib3.http2.extract_from_urllib3()
+
+    def test_http2_probe_blocked_per_thread(self) -> None:
+        assert http2_probe._values() == {}
+
+        connect_timeout = LONG_TIMEOUT
+        total_threads = 5
+        urllib3.http2.inject_into_urllib3()
+        try:
+
+            def make_request(_: typing.Any) -> tuple[float, float]:
+                with HTTPSConnectionPool(
+                    TARPIT_HOST,
+                    self.port,
+                    ca_certs=DEFAULT_CA,
+                    timeout=connect_timeout,
+                ) as pool:
+                    start_time = time.time()
+                    with pytest.raises(ConnectTimeoutError):
+                        pool.request("GET", "/", retries=False)
+                    end_time = time.time()
+                    return start_time, end_time
+
+            threadpool = concurrent.futures.ThreadPoolExecutor(total_threads)
+            values = list(threadpool.map(make_request, range(total_threads)))
+
+            min_start_time = min(start_time for start_time, _ in values)
+            max_start_time = max(start_time for start_time, _ in values)
+            assert (
+                max_start_time - min_start_time < connect_timeout
+            )  # Everything starts at a similar time.
+
+            # But all end times are sorted by completion and the differences between
+            # end times is greater or equal to the timeout value.
+            assert sorted(end_time for _, end_time in values) == [
+                end_time for _, end_time in values
+            ]
```

Comment:
As CI shows, I'm afraid this is not that simple. Thread scheduling (especially in crowded CI environment) can be completely arbitrary.